packages/mobile/src/store/offline-downloads/slice.ts

```diff
@@ -315,6 +319,11 @@ const slice = createSlice({
         state.collectionStatus[id] = OfflineDownloadStatus.ERROR
       } else if (type === 'track') {
         state.trackStatus[id] = OfflineDownloadStatus.ERROR
+        // re-queue the track
+        state.offlineQueue.push({
+          ...action.payload,
+          requeueCount: action.payload.requeueCount + 1
+        })
```

Comment:
Feels like the check for max_retry_count could happen here instead. Thoughts?

The reason I put it in the worker is so that we can use the proper `abandonJob` action, but I could also see setting it to `ABANDONED` state being a correct result of calling `errorJob`

Reply:
yeah i agree with your thinking, could also maybe happen in the "processNextJob" main function?